### DISCUSSION TITLE
Memray Live view shows "??? at ???" for Location

### DISCUSSION THREAD

**Discussion by JLowe-N** [bloomberg/memray#446] [Q&A] [answered]
Hi,

I am using Memray to attempt to profile a web application running on Tornado framework.
I have had recent success with tracking down excessive memory allocations within my code for some of my endpoints, but in profiling one bit of functionality - the reported location seems to be masked.

![image](https://github.com/bloomberg/memray/assets/37968405/8c5f7851-980c-4805-b8ae-21b933ff476d)

Is there any reason this memory allocation might not be visible?

Thank you for making such a great tool - it has been useful in troubleshooting a few web application issues so far!

**Comment by godlygeek** [comment]
The only reason I can think of for this, short of a bug in Memray itself, is that the memory might be allocated by an extension module from a native thread with no Python stack. If there's no Python stack then there's no Python location to show you, and we're stuck with just a `???`. If that's the case, running with `--native` should give you more info.

**Comment by JLowe-N** [comment, marked as answer]
Appreciate the feedback - this was a mistake on my side.
I realized that the code was still instrumented with cProfile using the `runCtx` feature --- essentially a large anonymous function.
I will try that native function if this happens again.
Removing cProfiler and running the live view again resolved it and I was able to trace the large allocations.
There was an inefficient eager load in SQL Alchemy that was pulling back GB worth of data.

**Reply by godlygeek** [reply]
Ah, yes, that makes sense! We monitor the Python stack using profiling hooks, so if you uninstall our profile hooks and install the cprofile ones instead, we lose the ability to associate allocations with a Python stack.
